version: '3'

services:
  jupyterhub:
    build: jupyterhub
    image: jupyterhub_img
    container_name: jupyterhub
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock
      - jupyterhub_data:/srv/jupyterhub
      - /etc:/jup_etc/
    environment:
      - DOCKER_JUPYTER_CONTAINER=jupyterlab_img
      - DOCKER_NETWORK_NAME=${COMPOSE_PROJECT_NAME}_default
      - HUB_IP=jupyterhub
      - HOST
      - IP
    labels:
      - "traefik.enable=true"
      - "traefik.frontend.rule=Host:${HOST}"
    restart: on-failure

  jupyterlab:
    build: jupyterlab
    image: jupyterlab_img
    container_name: jupyterlab-throaway
    network_mode: none
    command: echo

  reverse-proxy:
    image: traefik:1.7.17
    container_name: reverse-proxy
    ports:
      - "80:80"
      - "443:443"
      - "8080:8080"
    volumes:
      - ./reverse-proxy/traefik.toml:/etc/traefik/traefik.toml
      - /var/ssl:/var/ssl
      - /var/run/docker.sock:/var/run/docker.sock
    restart: on-failure

volumes:
  jupyterhub_data:
